File: Scripts/newTask/inheritance-single/javascript/inheritance-single.js
function Animal() {
    // ...
}

function Dog() {
    // ...
}

function Lab() {
    // ...
}

function Collie() {
    // ...
}

function Cat() {
    // ...
}

Animal.prototype.speak = function() {console.log("Animal speaking")};
Animal.prototype.walk = function() {console.log("Animal walking")};
Dog.prototype.speak = function() {console.log("Dog barking")};
Lab.prototype.speak = function() {console.log("Lab barking")};
Collie.prototype.speak = function() {console.log("Collie barking")};
Cat.prototype.speak = function() {console.log("Cat Meawing")};

var lab = new Dog();
lab.speak();  // shows "an animal makes a sound"
//lab.walk();

File: Scripts/volatile_and_force_compiler_task/inheritance-single/r/inheritance-single.r
Animal <- function(a) {
	value <- list(name = a)
	attr(value, "class") <- "Animal"
	value
}

speak <- function(obj) {
	UseMethod("speak")
}

speak.Animal <- function(obj) {
	cat (obj$name ,"speaking\n")
}

#Generic Function
walk <- function(obj) {
	UseMethod("walk")
}

walk.Animal <- function(obj) {
	cat("Animal walking\n")
}

speak.Dog <- function(obj) {
	cat (obj$name, " barking\n")
}


for ( i in 0:1000000 ) {
	doggy <- list(name="Dog")

	#inherit
	class(doggy) <- c("Dog","Animal")
	speak(doggy)
	walk(doggy)
}
